smart-contracts/hardhat.config.ts

```diff
@@ -1,24 +1,34 @@
-import '@openzeppelin/hardhat-upgrades'
+import * as dotenv from "dotenv"
+import {HardhatUserConfig} from "hardhat/config"
 import "@nomiclabs/hardhat-ethers"
-import "@typechain/hardhat"
+import '@openzeppelin/hardhat-upgrades'
 import "reflect-metadata"; // needed by tsyringe
-import {HardhatUserConfig} from "hardhat/config";
+import "@typechain/hardhat"
+
 // require('solidity-coverage');
 // require("hardhat-gas-reporter");
 // require('hardhat-contract-sizer');
 
-const alchemyUrl = process.env["ALCHEMY_URL"] ?? "no url specified"
+const envconfig = dotenv.config()
+
+const mainnetUrl = process.env["MAINNET_URL"] ?? "https://example.com"
+const ropstenUrl = process.env['ROPSTEN_URL'] ?? "https://example.com"
+const ropstenProxyAdminKey = process.env['ROPSTEN_PROXY_ADMIN_PRIVATE_KEY'] ?? "0xabcd"
 
 const config: HardhatUserConfig = {
     networks: {
         hardhat: {
             allowUnlimitedContractSize: false,
             chainId: 1,
             forking: {
-                url: alchemyUrl,
+                url: mainnetUrl,
                 blockNumber: 12865480,
             }
         },
+        ropsten: {
```

Comment:
We should add another config right here just called `mainnet` that uses the mainnnet private key and url.

Reply:
> We should add another config right here just called `mainnet` that uses the mainnnet private key and url.

done